client/src/js/subtraction/components/Detail.js

```diff
@@ -39,18 +61,19 @@ export class SubtractionDetail extends React.Component {
 
         const detail = this.props.detail;
 
+        const hasFiles = get(this.props, "files.length", 0) > 0;
```

Comment:
Instead, render the files `BoxGroup` but show a none found message using the `NoneFound` base components.

Did you have a situation when `files` was undefined?

Reply:
`files` was undefined was during testing when a shallow copy of `SubtractionDetail` is being rendered. Using `get` prevented that test from failing and works as a more defensive solution to directly access the file's length.

Reply:
Provide a list of fake file objects during the test. Having an undefined `files` field is an error condition and we need that to be thrown and reported to Sentry if it ever happens. Using `get` makes that potential bug silent.